src/client/auth/ticketZKProofMulti.ts

```diff
@@ -0,0 +1,148 @@
+import { AbstractAuthentication, AuthenticationMethodMulti, AuthenticationResult } from './abstractAuthentication'
+import { OffChainTokenConfig, MultiTokenInterface } from '../interface'
+import { OutletAction, Messaging } from '../messaging'
+import { Authenticator } from '@tokenscript/attestation'
+import { SignedUNChallenge } from './signedUNChallenge'
+import { UNInterface } from './util/UN'
+import { LocalOutlet } from '../../outlet/localOutlet'
+import { OutletIssuerInterface } from '../../outlet'
+import { createIssuerHashArray } from '../../utils'
+import { shouldUseRedirectMode } from '../../utils/support/getBrowserData'
+import { EasZkProof } from '@tokenscript/attestation/dist/eas/EasZkProof'
+import { DEFAULT_EAS_SCHEMA, TokenType } from '../../outlet/ticketStorage'
+import { EAS_RPC_CONFIG } from '../../core/eas'
+
+export class TicketZKProofMulti extends AbstractAuthentication implements AuthenticationMethodMulti {
+	TYPE = 'ticketZKProofMulti'
+
+	private messaging = new Messaging()
+
+	// NOTE: A limitation at this time is that the user can only authenticate tokens
+	// that use the same issuer origin, email and it's underlying config (see: firstCollectionFound via getTokenProofMulti).
+
+	async getTokenProofMulti(userTokens: MultiTokenInterface[], request: any): Promise<AuthenticationResult> {
+		const firstCollectionFound = Object.keys(userTokens)[0]
+		const tokenCollectionConfig = userTokens[firstCollectionFound].requestTokens[0]
+		const issuerCollectionConfig = userTokens[firstCollectionFound].issuerConfig
+		let redirectMode: false | string = this.getRedirectModeSetting(tokenCollectionConfig, request)
+		let useEthKey: UNInterface | null = null
+		if (issuerCollectionConfig?.unEndPoint) {
+			let unChallenge = new SignedUNChallenge(this.client)
+			request.options = {
+				...request.options,
+				unEndPoint: issuerCollectionConfig?.unEndPoint,
+			}
+			let unRes = await unChallenge.getTokenProof(request)
+			useEthKey = unRes.data as UNInterface
+		}
+		const useEthKeyAddress = useEthKey ? useEthKey.address : ''
+		const address = request.address ? request.address : useEthKeyAddress
+		const wallet = request.wallet ? request.wallet : ''
+		let output
+		if (new URL(issuerCollectionConfig.tokenOrigin).origin === window.location.origin) {
+			output = await this.authenticateLocally(userTokens, issuerCollectionConfig, address, wallet, redirectMode)
```

Comment:
@nicktaras could you make same params order for authenticateLocally(userTokens, issuerCollectionConfig...) and authenticateCrossOrigin(issuerCollectionConfig, userTokens...) to avoid problems in future?